src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -105,9 +105,9 @@ export class TableOfContents implements OnInit, AfterViewInit, OnDestroy {
   }
 
   addHeaders(sectionName: string, docViewerContent: HTMLElement) {
-    const headers = docViewerContent.querySelectorAll('h3, h4');
+    const headers = Array.from(docViewerContent.querySelectorAll('h3, h4')) as HTMLHeadingElement[];
```

Comment:
It's pretty much the same, but I think you can do this which is a little shorter `Array.from<HTMLHeadingElement>(docViewerContent.querySelectorAll('h3, h4'))`.